src/simplewallet/simplewallet.cpp

```diff
@@ -4468,6 +4485,23 @@ bool simple_wallet::show_transfer(const std::vector<std::string> &args)
       success_msg_writer() << "Timestamp: " << get_human_readable_timestamp(pd.m_timestamp);
       success_msg_writer() << "Amount: " << print_money(pd.m_amount);
       success_msg_writer() << "Payment ID: " << payment_id;
+      if (pd.m_unlock_time < CRYPTONOTE_MAX_BLOCK_NUMBER)
```

Comment:
The code uses the max of unlock time and tx block height, so it will show the correct unlock time (except if that tx actually had an unlock time in the future, but those are probably very rare). The default unlock time on the chain is 0 if there is none.